firebase-ai/src/main/kotlin/com/google/firebase/ai/TemplateChat.kt

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.ai
+
+import android.graphics.Bitmap
+import com.google.firebase.ai.type.Content
+import com.google.firebase.ai.type.GenerateContentResponse
+import com.google.firebase.ai.type.ImagePart
+import com.google.firebase.ai.type.InlineDataPart
+import com.google.firebase.ai.type.InvalidStateException
+import com.google.firebase.ai.type.PublicPreviewAPI
+import com.google.firebase.ai.type.TextPart
+import com.google.firebase.ai.type.content
+import java.util.LinkedList
+import java.util.concurrent.Semaphore
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.flow.onEach
+
+/**
+ * Representation of a multi-turn interaction with a model.
+ *
+ * Captures and stores the history of communication in memory, and provides it as context with each
+ * new message.
+ *
+ * **Note:** This object is not thread-safe, and calling [sendMessage] multiple times without
+ * waiting for a response will throw an [InvalidStateException].
+ *
+ * @param model The model to use for the interaction.
+ * @param templateID The template ID for this chat session
+ * @property history The previous content from the chat that has been successfully sent and received
+ * from the model. This will be provided to the model for each message sent (as context for the
+ * discussion).
+ */
+@PublicPreviewAPI
+public class TemplateChat(
```

Comment:
Do we want this class to have a public constructor? or do we want it to be created only when calling `startChat()`?